### PR TITLE
Redux3

### PR DESCRIPTION
**Pull request by tbuschto**

### CODE REVIEW

File: src/decorators/connect.ts [hunk 1]
```diff
@@ -62,18 +65,36 @@ function updateConfig(target: Constructor<any>, config: Connection) {
...
 
-function addInjectorInjection(proxy: Constructor<any>, config: Connection) {
-  if (config.stateMapper) {
+function apply(options: ConnectedCreationOptions) {
+  const connection: Connection = Reflect.getOwnMetadata(componentConfigKey, options.proxy);
+  const target = Reflect.apply(options.target, options.context, options.creationArgs);
+  return connectTarget(target, connection, options);
+}
+
+function connectTarget(
```
Comment (cpetrov) [inline src/decorators/connect.ts:84]: Maybe inline this function? It seems to be used only at one place and the parameter declaration is almost as long as the function itself

File: test/connect.spec.tsx [hunk 2]
```diff
@@ -74,7 +75,22 @@ describe('connect', () => {
       );
     });
 
-    it('implies @component', () => {
+    it('can be combined with @component before', () => {
```
Comment (cpetrov) [inline test/connect.spec.tsx:78]: before what?

File: test/connect.spec.tsx [hunk 3]
```diff
@@ -74,7 +75,22 @@ describe('connect', () => {
...
+      instance.append(<TextView/>);
+      expect(instance.children().length).to.equal(0);
+    });
+
+    it('can be combined with @component after', () => {
```
Comment (cpetrov) [inline test/connect.spec.tsx:83]: after what?

File: src/decorators/connect.ts (hunk 1 above)
Comment (tbuschto) [reply src/decorators/connect.ts:84]: Was supposed to be used twice. Now it is.

File: test/connect.spec.tsx (hunk 3 above)
Comment (tbuschto) [reply test/connect.spec.tsx]: fixed

**Pull request merged**